ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -1045,39 +1036,20 @@ private void computeTargetViewForMaintainVisibleContentPosition() {
     }
 
     int currentScrollY = getScrollY();
-    int min = mMaintainVisibleContentPositionData.minIndexForVisible;
-    int max = contentView.getChildCount();
-    while (min <= max) {
-      int mid = min + ((max - min) / 2);
-      View child = contentView.getChildAt(mid);
-
-      // If the start of this item is less than the current scroll, it's above of the visible frame
-      if (child.getY() < currentScrollY) {
-        min = mid + 1;
-        continue;
-      }
-
-      // If it's not visible at all, it is below the visible frame
-      if (!child.isShown()) {
-        max = mid;
-        continue;
-      }
-
-      View prevChild = contentView.getChildAt(mid - 1);
-      if (prevChild != null && prevChild.getGlobalVisibleRect(new Rect())) {
-        max = mid;
-        continue;
+    int minIndex = mMaintainVisibleContentPositionData.minIndexForVisible;
```

Comment:
I had some issues with binary search infinite looping so I instead of debugging it I just ported the iOS logic pretty much line for line. I think this should be ok.